backend/app/requests/adequacy.py

```diff
@@ -82,11 +85,56 @@ def adequacy_request(app, flask_request, engine):
         measurer_name = config.get('measurer')['haversine']
 
     # Exit early if there is no data.
-    if provider_addresses and service_area_ids:
+    if not (provider_locations and service_area_ids):
+        return []
+    # If caching is enabled, return cached data.
+    elif dataset_hint and config.get('cache_adequacy_requests'):
+        return _get_cached_adequacy_response(
+            dataset_hint=dataset_hint,
+            measurer_name=measurer_name,
+            service_area_ids=service_area_ids,
+            provider_locations=provider_locations,
```

Comment:
Can we use similar args names? provider_locations / locations?

Reply:
I do not understand.

Reply:
I now understand. Done.